app/controllers/admin/themes_controller.rb

```diff
@@ -408,6 +425,8 @@ def update_settings
   def update_translations
     return unless target_translations = theme_params[:translations]
 
+    I18n.locale = theme_params[:locale] if theme_params[:locale].present?
```

Comment:
A validation would be useful here too. And a corresponding update to the existing specs (i.e. add a case for updating translations with a locale param).